include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -56,30 +56,31 @@ MultigridPreconditionerBase<dim, Number>::MultigridPreconditionerBase(MPI_Comm c
 template<int dim, typename Number>
 void
 MultigridPreconditionerBase<dim, Number>::initialize(
-  MultigridData const &                       data,
-  dealii::Triangulation<dim> const *          tria,
-  dealii::FiniteElement<dim> const &          fe,
-  std::shared_ptr<dealii::Mapping<dim> const> mapping,
-  bool const                                  operator_is_singular,
-  Map const &                                 dirichlet_bc,
-  PeriodicFacePairs const &                   periodic_face_pairs)
+  MultigridData const &                                                  data,
+  dealii::Triangulation<dim> const *                                     tria,
+  std::vector<std::shared_ptr<dealii::Triangulation<dim> const>> const & coarse_triangulations,
+  dealii::FiniteElement<dim> const &                                     fe,
+  std::shared_ptr<dealii::Mapping<dim> const>                            mapping,
+  bool const                                                             operator_is_singular,
+  Map const &                                                            dirichlet_bc,
+  PeriodicFacePairs const &                                              periodic_face_pairs)
 {
   this->data = data;
 
   this->triangulation = tria;
 
+  this->coarse_triangulations = coarse_triangulations;
```

Comment:
calling here `dealii::MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence()` instead of the assignment of the vector of shared pointers fixes the bug introduced in this PR (see issue #303).

Reply:
Is the copy of a std::vector of shared pointers problematic?